### PR TITLE
No Bug: Update Github Actions Xcode version to 12.

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #<number>

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by iccub** [comment]
Tests fail due to CoreNFC crash on xcode12, need to wait for an update

**Comment by kylehickinson** [comment]
@iccub We can continue with this one but use `12.1` instead which is now available on GH actions

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -14,9 +14,9 @@ jobs:
     runs-on: macOS-latest
     env:
         # The XCode version to use. If you want to update it please refer to this document:
-        # https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners#xcode
+        # https://docs.github.com/en/actions/reference/specifications-for-github-hosted-runners#supported-software
         # and set proper version.
-        XCODE_VERSION: 11.5
+        XCODE_VERSION: 12.2
```
Comment (kylehickinson) [inline .github/workflows/build_and_test.yml]: Use `12.1` instead which is now available on GH Actions

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (iccub) [reply .github/workflows/build_and_test.yml]: isn't it going to fail due to the missing corenfc framework bug?

Comment (kylehickinson) [reply .github/workflows/build_and_test.yml]: The CoreNFC bug is fixed in the 12.1 GM :) At least I haven't had a problem with it locally

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**